runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java

```diff
@@ -662,12 +789,30 @@ public void onGameStateChanged(GameStateChanged gameStateChanged)
 	{
 		switch (gameStateChanged.getGameState())
 		{
+			case LOGGING_IN:
+				loggingIn = true;
+				break;
+			case LOADING:
+				if (!loggingIn)
+				{
+					updateTzhaarTimerOnLoading();
+				}
+				break;
 			case HOPPING:
+				loggingIn = true;
 			case LOGIN_SCREEN:
+				removeTzhaarTimer();
+				saveTzhaarConfig();
 				removeTbTimers();
 				break;
 			case LOGGED_IN:
 				loggedInRace = true;
+				if (loggingIn)
+				{
+					loggingIn = false;
+					loadTzhaarConfig();
```

Comment:
What's the purpose of this load-into-memory-then-wipe-from-config pattern you have going here? I think you can just use the values from config and simplify most of this?